### DISCUSSION TITLE
Is there any tools to check rego grammer?

### DISCUSSION THREAD

**Discussion by yaohy2014** [open-policy-agent/community#559] [Community]
Is there any tools to check rego grammer?

**Comment by anderseknert** [comment]
Hi there @yaohy2014! There are a few options, depending on what you want:

* You can use the `opa check` command to find syntax errors in your Rego files.
* Use `opa check --strict` to enable even more checks, like unused variables, etc.
* Use the linter [Regal](https://github.com/styraInc/regal) for checking your Rego files not just for errors, but for advice on best practices, style, performance and much more. 

Let me know if there's anything you're missing or would like to know more about.

**Reply by yaohy2014** [reply]
Thanks a lot. 
Is there any libararies? I want to call a function to check the rego file or rego files in a directory.

**Reply by anderseknert** [reply]
You can use OPA itself as a Go library. Just see the code for the `check` command if you want to do something like that programmatically.
